### PR TITLE
Nat/create with rg

### LINKED ISSUE #471: in API, support the input of resource group in createFunctionApp()
Currently, the API of create Function App is as follow:

const functionAppId: string = await vscode.commands.executeCommand<string>('azureFunctions.createFunctionApp', '00000000-0000-0000-0000-000000000000');

Could you provide one more input parameter of resource group name?

### PR DESCRIPTION
**Pull request by nturinski**
Fixes #471 
Allow resourceGroup to be a programmatic property for Function apps

### CODE REVIEW

File: src/commands/createFunctionApp.ts [hunk 1]
```diff
@@ -4,16 +4,23 @@
...
         node = <IAzureParentNode>await tree.showNodePicker(AzureTreeDataProvider.subscriptionContextValue);
     } else {
-        node = arg;
+        node = subscription;
+    }
+    if (resourceGroup) {
+        if (typeof resourceGroup !== 'string') {
+            resourceGroup = resourceGroup.treeItem.client.resourceGroup;
+        }
+        actionContext.properties.resourceGroup = resourceGroup;
```
Comment (EricJizbaMSFT) [inline src/commands/createFunctionApp.ts]: `actionContext.properties` is used for telemetry and we don't want to collect this in telemetry

We'll have to find a different way to pass the resource group along

File: src/commands/createFunctionApp.ts [hunk 2]
```diff
@@ -4,16 +4,23 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { AzureTreeDataProvider, IActionContext, IAzureNode, IAzureParentNode } from 'vscode-azureextensionui';
+import { FunctionAppTreeItem } from '../tree/FunctionAppTreeItem';
 import { nodeUtils } from '../utils/nodeUtils';
 
-export async function createFunctionApp(actionContext: IActionContext, tree: AzureTreeDataProvider, arg?: IAzureParentNode | string): Promise<string> {
+export async function createFunctionApp(actionContext: IActionContext, tree: AzureTreeDataProvider, subscription?: IAzureParentNode | string, resourceGroup?: IAzureParentNode<FunctionAppTreeItem> | string): Promise<string> {
```
Comment (EricJizbaMSFT) [inline src/commands/createFunctionApp.ts]: When will resourceGroup ever be `IAzureParentNode<FunctionAppTreeItem>`? Shouldn't it just be string?

File: src/commands/createFunctionApp.ts (hunk 1 above)
Comment (nturinski) [reply src/commands/createFunctionApp.ts]: There are UserSettings here:
https://github.com/Microsoft/vscode-azurefunctions/blob/56a097178f1deac96f69cac174be26f2c2899825/src/commands/createFunction/createFunction.ts#L150

But it seems like it's used for appSettings which get passed into the ActionContext.  Should we just create a new property?

File: src/commands/createFunctionApp.ts (hunk 2 above)
Comment (nturinski) [reply src/commands/createFunctionApp.ts]: I wasn't sure how it was going to be used, but I thought a user could just pass another function app and use the same resource group as the one passed.

File: src/commands/createFunctionApp.ts [hunk 3]
```diff
@@ -6,16 +6,16 @@
...
+        node = await nodeUtils.getSubscriptionNode(tree, subscription);
+    } else if (!subscription) {
         node = <IAzureParentNode>await tree.showNodePicker(AzureTreeDataProvider.subscriptionContextValue);
     } else {
-        node = arg;
+        node = subscription;
     }
 
-    const funcAppNode: IAzureNode = await node.createChild(actionContext);
+    const funcAppNode: IAzureNode = await node.createChild(actionContext, resourceGroup);
```
Comment (EricJizbaMSFT) [inline src/commands/createFunctionApp.ts]: createChild only supports one `userOptions` parameter as of right now. You can combine them into one object: `{ actionContext, resourceGroup }`

**Pull request merged**